sogs/model/filter.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+from .. import config, crypto, db, utils, session_pb2 as protobuf
+import random
+
+from .. import crypto, db, config
+from ..db import query
+from ..hashing import blake2b
+from ..web import app
+from nacl.signing import SigningKey
+from .exc import NoSuchUser, BadPermission, PostRejected
+from sogs.model.user import User
+from sogs.model.bot import Bot
+from sogs.model.room import Room, alphabet_filter_patterns
```

Comment:
Should try get this Filter to exist separately from these Models.

Like it just gets fed a raw text message checks if its naughty according to its configured filters. Then returns a bool if its safe. 

That way this can be extracted out and used by the bots when the bots themselves have no access to the models

Reply:
agreed. we can turn it into a basic bot to use for testing